drivers/ukintctlr/Makefile.uk

```diff
@@ -6,4 +6,5 @@
 
 UK_DRIV_INTCTLR_BASE := $(UK_DRIV_BASE)/ukintctlr
 
+$(eval $(call _import_lib,$(UK_DRIV_INTCTLR_BASE)/xpic))
```

Comment:
That I understood, what I am saying is that it looks vague and IMO every irq controller driver should maybe have its own directory with its corresponding C/header source files. This way you could also have more specific directories. I find `xpic` too vague.

Reply:
@mogasergiu, @mschlumpp I have no opinion about the name, especially given that I don't maintain this driver. I only picked it because I assumed that PIC, APIC, and x2APIC would have inter-dependencies and would likely live under the same driver. If you can think of something more suitable, I would be more than happy to update 🙃

Reply:
I just think each irq controller should have its own directory: `i8259` for `PIC`, `apic` for `APIC`, `gic-v2` for `GICv2`, `gic-v3` for `GICv3`, etc. Each directory having the driver's source code as well as its corresponding private header (containing registers and other specific stuff that should not be accessible unless accessed through the driver's methods). E.g.
```
ukintctlr/
       i8259/
           i8259.c
           i8259.h
       apic/
           apic.c
           apic.h
       gic-v2/
           gic-v2.c
           gic-v2.h
       gic-v3/
           gic-v3.c
           gic-v3.h
```

Reply:
@mogasergiu Yes, but: the current state of the x86 interrupt system is that APIC+PIT are bundled together. You have to use both if you want to use the APIC. This PR is mostly "move everything in the new hierarchy" and not "implement proper APIC support".

Reply:
> @mogasergiu Yes, but: the current state of the x86 interrupt system is that APIC+PIT are bundled together. You have to use both if you want to use the APIC. This PR is mostly "move everything in the new hierarchy" and not "implement proper APIC support".

Agreed. Regarding `xPIC` if no one comes up with a better name I'll leave as is and let you folks change it if you decide to do so later. Notice however that this would involve a non-backwards compatible change on the Kconfig options.